src/vma/util/flow_instrumentation.h

```diff
@@ -0,0 +1,199 @@
+#endif
+
+#ifdef FLOW_RX_INPUT_CB_UDP_WRAPPER
+	#define INSTRUMENT_START_RX_INPUT_CB_UDP_WRAPPER	VMA_TIME_IBPROF_START(88, "rx_input_cb(udp)");
+        #define INSTRUMENT_END_RX_INPUT_CB_UDP_WRAPPER		VMA_TIME_IBPROF_END(88);
+#else
+        #define INSTRUMENT_START_RX_INPUT_CB_UDP_WRAPPER
+        #define INSTRUMENT_END_RX_INPUT_CB_UDP_WRAPPER
+#endif
+
+
+
+/*************************************************************************/
+
+#ifdef FLOW_SEND_TO_FASTSEND_TCP
+	#define INSTRUMENT_START_SEND_TO_FASTSEND_TCP	     VMA_TIME_IBPROF_START(101, "send to fast_send(tcp)");
+	#define INSTRUMENT_END_SEND_TO_FASTSEND_TCP	     VMA_TIME_IBPROF_END(101);
+#else
+       #define INSTRUMENT_START_SEND_TO_FASTSEND_TCP
+        #define INSTRUMENT_END_SEND_TO_FASTSEND_TCP
+
+#endif
+
+#ifdef FLOW_SEND_TO_FASTSEND_UDP
+        #define INSTRUMENT_START_SEND_TO_FASTSEND_UDP        VMA_TIME_IBPROF_START(102, "send to fast_send(udp)");
+        #define INSTRUMENT_END_SEND_TO_FASTSEND_UDP          VMA_TIME_IBPROF_END(102);
+#else
+	#define INSTRUMENT_START_SEND_TO_FASTSEND_UDP
+        #define INSTRUMENT_END_SEND_TO_FASTSEND_UDP
+
+#endif
+
+#ifdef FLOW_FASTSEND_TCP_TO_SIMPLE_SEND_RING_BUFFER
+	#define INSTRUMENT_START_FASTSEND_TCP_TO_SIMPLE_SEND_RING_BUFFER VMA_TIME_IBPROF_START(111, "fastsend(tcp) to send_ring_buffer");
+	#define INSTRUMENT_END_FASTSEND_TCP_TO_SIMPLE_SEND_RING_BUFFER   VMA_TIME_IBPROF_END(111);
+#else
+        #define INSTRUMENT_START_FASTSEND_TCP_TO_SIMPLE_SEND_RING_BUFFER
+        #define INSTRUMENT_END_FASTSEND_TCP_TO_SIMPLE_SEND_RING_BUFFER
+#endif
+
+#ifdef FLOW_FASTSEND_UDP_TO_SIMPLE_SEND_RING_BUFFER
+        #define INSTRUMENT_START_FASTSEND_UDP_TO_SIMPLE_SEND_RING_BUFFER VMA_TIME_IBPROF_START(112, "fastsend(udp) to send_ring_buffer");
+        #define INSTRUMENT_END_FASTSEND_UDP_TO_SIMPLE_SEND_RING_BUFFER   VMA_TIME_IBPROF_END(112); 
+#else
+        #define INSTRUMENT_START_FASTSEND_UDP_TO_SIMPLE_SEND_RING_BUFFER
+        #define INSTRUMENT_END_FASTSEND_UDP_TO_SIMPLE_SEND_RING_BUFFER
+#endif
+
+
+#ifdef FLOW_RING_SIMPLE_SEND_RING_BUFFER_WRAPPER
+        #define INSTRUMENT_START_RING_SIMPLE_SEND_RING_BUFFER_WRAPPER VMA_TIME_IBPROF_START(121, "ring_simple::send_ring_buffer()");
+        #define INSTRUMENT_END_RING_SIMPLE_SEND_RING_BUFFER_WRAPPER   VMA_TIME_IBPROF_END(121);
+#else
+        #define INSTRUMENT_START_RING_SIMPLE_SEND_RING_BUFFER_WRAPPER
+        #define INSTRUMENT_END_RING_SIMPLE_SEND_RING_BUFFER_WRAPPER
+#endif
+
+#ifdef FLOW_RING_SIMPLE_SEND_RING_BUFFER_TO_SEND
+        #define INSTRUMENT_START_RING_SIMPLE_SEND_RING_BUFFER_TO_SEND  VMA_TIME_IBPROF_START(131, "send_ring_buffer() to send");
+        #define INSTRUMENT_END_RING_SIMPLE_SEND_RING_BUFFER_TO_SEND    VMA_TIME_IBPROF_END(131);
+#else
+        #define INSTRUMENT_START_RING_SIMPLE_SEND_RING_BUFFER_TO_SEND
+        #define INSTRUMENT_END_RING_SIMPLE_SEND_RING_BUFFER_TO_SEND
+#endif
+
+
+
+
+#endif //FLOWINSTRUMENTATION
+
```

Comment:
@guy101 all of this macros have very long names. I do not see any obvious reason not to use short and simple names for this purpose. Anyone can look at prefix to understand that it relates profiling and read next line of the code. 
Please consider short names. You have description as FLOW_RING_SIMPLE_SEND_RING_BUFFER_TO_SEND and "send_ring_buffer() to send"
For example:
```
#ifdef FLOW_RING_SIMPLE_SEND_RING_BUFFER_TO_SEND
        #define INSTRUMENT_START_131  VMA_TIME_IBPROF_START(131, "send_ring_buffer() to send");
        #define INSTRUMENT_END_131     VMA_TIME_IBPROF_END(131);
#else
        #define INSTRUMENT_START_131
        #define INSTRUMENT_END_131
#endif
```